x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -171,84 +169,109 @@ export async function installTemplateForDataStream({
   });
 }
 
+interface TemplateMapEntry {
+  template:
+    | {
+        mappings: NonNullable<RegistryElasticsearch['index_template.mappings']>;
+      }
+    | {
+        settings: NonNullable<RegistryElasticsearch['index_template.settings']> | object;
+      };
+}
+type TemplateMap = Record<string, TemplateMapEntry>;
 function putComponentTemplate(
-  body: object | undefined,
-  name: string,
-  esClient: ElasticsearchClient
-): { clusterPromise: Promise<any>; name: string } | undefined {
-  if (body) {
-    const esClientParams = {
-      name,
-      body,
-    };
-
-    return {
-      // @ts-expect-error body expected to be ClusterPutComponentTemplateRequest
-      clusterPromise: esClient.cluster.putComponentTemplate(esClientParams, { ignore: [404] }),
-      name,
-    };
+  esClient: ElasticsearchClient,
+  params: {
+    body: TemplateMapEntry;
+    name: string;
+    create?: boolean;
   }
+): { clusterPromise: Promise<any>; name: string } {
+  const { name, body, create = false } = params;
+  return {
+    clusterPromise: esClient.cluster.putComponentTemplate(
+      // @ts-expect-error body is missing required key `settings`. TemplateMapEntry has settings *or* mappings
+      { name, body, create },
+      { ignore: [404] }
+    ),
+    name,
+  };
 }
 
-function buildComponentTemplates(registryElasticsearch: RegistryElasticsearch | undefined) {
-  let mappingsTemplate;
-  let settingsTemplate;
+const mappingsSuffix = '-mappings';
+const settingsSuffix = '-settings';
+const userSettingsSuffix = '-user_settings';
+type TemplateBaseName = string;
+type UserSettingsTemplateName = `${TemplateBaseName}${typeof userSettingsSuffix}`;
+
+const isUserSettingsTemplate = (name: string): name is UserSettingsTemplateName =>
+  name.endsWith(userSettingsSuffix);
+
+function buildComponentTemplates(
+  templateName: string,
+  registryElasticsearch: RegistryElasticsearch | undefined
+) {
+  const mappingsTemplateName = `${templateName}${mappingsSuffix}`;
+  const settingsTemplateName = `${templateName}${settingsSuffix}`;
+  const userSettingsTemplateName = `${templateName}${userSettingsSuffix}`;
+
+  const templatesMap: TemplateMap = {};
 
   if (registryElasticsearch && registryElasticsearch['index_template.mappings']) {
-    mappingsTemplate = {
+    templatesMap[mappingsTemplateName] = {
       template: {
-        mappings: {
-          ...registryElasticsearch['index_template.mappings'],
-        },
+        mappings: registryElasticsearch['index_template.mappings'],
       },
     };
   }
 
   if (registryElasticsearch && registryElasticsearch['index_template.settings']) {
-    settingsTemplate = {
+    templatesMap[settingsTemplateName] = {
       template: {
         settings: registryElasticsearch['index_template.settings'],
       },
     };
+
+    // return empty/stub template
+    templatesMap[userSettingsTemplateName] = {
+      template: {
+        settings: {},
+      },
+    };
```

Comment:
this should be moved outside of the L228 condition. we actually don't need to check on the existence of `index_template.settings` on the package information. even if that is not present, we should always install this custom template